test/fixtures/rules/std_rule_cases/L016.yml

```diff
@@ -156,3 +156,77 @@ test_long_jina_comment_ignore:
       max_line_length: 80
       L016:
         ignore_comment_lines: True
+
+
+test_for_loop:
+  # A Jinja for loop
+  pass_str: |
+    {% for elem in 'foo' %}
+    SELECT '{{ elem }}' FROM table1;
+    SELECT '{{ elem }}' FROM table2;
+    {% endfor %}
+
+
+test_for_loop_repeating_elements_starts_with_literal:
+  # A Jinja for loop with repeating elements (that are difficult to match)
+  # but starting with a literal that can be used to match
+  pass_str: |
+    {% set elements = 'foo' %}
+    SELECT
+        CASE
+            {% for elem in elements %}
+            WHEN '{{ elem }}' = '' THEN 1
+            WHEN '{{ elem }}' = '' THEN 1
+            {% endfor %}
+        END
+
+
+test_for_loop_starting_with_templated_piece:
+  # A Jinja for loop starting with non-literals
+  # But unique parts can be used to match
+  pass_str: |
+    {% set elements = 'foo' %}
+    {% set when = 'WHEN' %}
+    SELECT
+        CASE
+            {% for elem in elements %}
+            {{ when }} '{{ elem }}' = '' THEN 1
+            {{ when }} '{{ elem }}' = '' THEN 2
+            {% endfor %}
+        END
+
+test_for_loop_fail_complex_match:
+  # Ideally this should pass, but jinja templating
+  # fails to handle this as difficult to match.
```

Comment:
Is it worth noting that we consider this code potentially unrealistic due to the repetition (but still possible)?